### PR TITLE
[Cache] Add Redis Relay doc

### LINKED ISSUE #17813: [Cache] Add Redis Relay support
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#48930
| PR author(s) | @ostrolucky
| Merged in    | 6.3

### PR DESCRIPTION
**Pull request by ostrolucky**
Closes #17813. Let me know if this is enough or what else should be done.

### CODE REVIEW

File: components/cache/adapters/redis_adapter.rst [hunk 1]
```diff
@@ -258,6 +258,7 @@ Read more about this topic in the official `Redis LRU Cache Documentation`_.
 .. _`Redis`: https://github.com/phpredis/phpredis
 .. _`RedisArray`: https://github.com/phpredis/phpredis/blob/master/arrays.markdown#readme
 .. _`RedisCluster`: https://github.com/phpredis/phpredis/blob/master/cluster.markdown#readme
+.. _`Relay`: https://relay.so/
```
Comment (tillkruss) [inline components/cache/adapters/redis_adapter.rst:269]: Might be nice to link to GitHub instead:

```suggestion
.. _`Relay`: https://github.com/cachewerk/relay
```

Comment (ostrolucky) [reply components/cache/adapters/redis_adapter.rst:269]: I don't really agree. If you want to go to documentation, GH repo will redirect you to relay.so anyways. I know you are following other, above listed clients, but none of them have a project website. But let's see if there is same feedback by other people as well.

Comment (tillkruss) [reply components/cache/adapters/redis_adapter.rst:269]: Ah, right. Maybe `https://relay.so/docs`?

**Comment by OskarStark** [comment]
This one should target 6.3 branch

File: session.rst [hunk 2]
```diff
@@ -503,7 +503,7 @@ a Symfony service for the connection to the Redis server:
                     # - { 'prefix': 'my_prefix', 'ttl': 600 }
 
             Redis:
-                # you can also use \RedisArray, \RedisCluster or \Predis\Client classes
+                # you can also use \RedisArray, \RedisCluster, \Relay\Relay (since Symfony 6.1) or \Predis\Client classes
```
Comment (OskarStark) [inline session.rst]:
```suggestion
                # you can also use \RedisArray, \RedisCluster, \Relay\Relay or \Predis\Client classes
```

File: components/lock.rst [hunk 3]
```diff
@@ -603,7 +603,7 @@ RedisStore
 ~~~~~~~~~~
 
 The RedisStore saves locks on a Redis server, it requires a Redis connection
-implementing the ``\Redis``, ``\RedisArray``, ``\RedisCluster`` or
+implementing the ``\Redis``, ``\RedisArray``, ``\RedisCluster``, \Relay\Relay (since Symfony 6.1) or
```
Comment (OskarStark) [inline components/lock.rst]:
```suggestion
implementing the ``\Redis``, ``\RedisArray``, ``\RedisCluster``, ``\Relay\Relay`` or
```

File: components/lock.rst [hunk 4]
```diff
@@ -603,7 +603,7 @@ RedisStore
 ~~~~~~~~~~
```
Comment (OskarStark) [inline components/lock.rst:604]:
```suggestion

.. versionadded:: 6.1

    ``\Relay\Relay`` support was introduced in Symfony 6.1.

```

File: components/cache/adapters/redis_adapter.rst (hunk 1 above)
Comment (OskarStark) [reply components/cache/adapters/redis_adapter.rst:269]: Let's use the project website 👍

File: components/cache/adapters/redis_adapter.rst [hunk 5]
```diff
@@ -166,9 +166,9 @@ Available Options
 ~~~~~~~~~~~~~~~~~
```
Comment (OskarStark) [inline components/cache/adapters/redis_adapter.rst:171]:
```suggestion

.. versionadded:: 6.1

    Using ``\Relay\Relay`` for ``class`` was introduced in Symfony 6.1.

```

File: components/cache/adapters/redis_adapter.rst [hunk 6]
```diff
@@ -25,9 +25,9 @@ to utilize a cluster of servers to provide redundancy and/or fail-over is also a
 
     **Requirements:** At least one `Redis server`_ must be installed and running to use this
     adapter. Additionally, this adapter requires a compatible extension or library that implements
-    ``\Redis``, ``\RedisArray``, ``RedisCluster``, or ``\Predis``.
+    ``\Redis``, ``\RedisArray``, ``RedisCluster``, ``\Relay\Relay`` or ``\Predis``.
 
-This adapter expects a `Redis`_, `RedisArray`_, `RedisCluster`_, or `Predis`_ instance to be
+This adapter expects a `Redis`_, `RedisArray`_, `RedisCluster`_, `Relay`_ (since Symfony 6.1) or `Predis`_ instance to be
 passed as the first parameter. A namespace and default cache lifetime can optionally be passed
 as the second and third parameters::
```
Comment (OskarStark) [inline components/cache/adapters/redis_adapter.rst:33]: This must go in line 49 (above the headline `Configure the Connection`):
```
.. versionadded:: 6.1

    Support for `Relay`_ was introduced in Symfony 6.1.
```

File: components/cache/adapters/redis_adapter.rst [hunk 7]
```diff
@@ -25,9 +25,9 @@ to utilize a cluster of servers to provide redundancy and/or fail-over is also a
 
     **Requirements:** At least one `Redis server`_ must be installed and running to use this
     adapter. Additionally, this adapter requires a compatible extension or library that implements
-    ``\Redis``, ``\RedisArray``, ``RedisCluster``, or ``\Predis``.
+    ``\Redis``, ``\RedisArray``, ``RedisCluster``, ``\Relay\Relay`` or ``\Predis``.
 
-This adapter expects a `Redis`_, `RedisArray`_, `RedisCluster`_, or `Predis`_ instance to be
+This adapter expects a `Redis`_, `RedisArray`_, `RedisCluster`_, `Relay`_ (since Symfony 6.1) or `Predis`_ instance to be
```
Comment (OskarStark) [inline components/cache/adapters/redis_adapter.rst]:
```suggestion
This adapter expects a `Redis`_, `RedisArray`_, `RedisCluster`_, `Relay`_ or `Predis`_ instance to be
```

**Review by OskarStark — APPROVED** [review]
Thank you!

**Comment by javiereguiluz** [comment]
Thanks Gabriel.

**Pull request merged**